NU1603 Warnings

<!--
GitHub is reserved for bug reports and feature requests. The best place
to ask a general question is at the Elastic Discourse forums at
https://discuss.elastic.co. If you are in fact posting a bug report or
a feature request, please include one and only one of the below blocks
in your new issue.
-->

<!--
If you are filing a bug report, please remove the below feature
request block and provide responses for all of the below items.
-->

**NEST/Elasticsearch.Net version**:
5.4/5.5

**Description of the problem including expected versus actual behavior**:
```
warning NU1603: NEST 5.4.0 depends on Newtonsoft.Json (>= 10.0.0 && < 11.0.0) but Newtonsoft.Json 10.0.0 was not found. An approximate best match of Newtonsoft.Json 10.0.1 was resolved.
```
Shouldn't have these warnings when build in latest vs2017

**Steps to reproduce**:
 1. Updated VS2017 to Version 15.3.0
 2. Build the project having reference on NEST 5.4/5.5

Same problem with versions 2.4.6/2.4.6.

> error NU1603: NEST 2.4.6 depends on Newtonsoft.Json (>= 9.0.0 && < 10.0.0) but Newtonsoft.Json 9.0.0 was not found. An approximate best match of Newtonsoft.Json 9.0.1 was resolved.

--- a/build/scripts/Paths.fsx
+++ b/build/scripts/Paths.fsx
@@ -34,6 +34,15 @@ module Paths =
     let Output(folder) = sprintf "%s/%s" BuildOutput folder
     let Source(folder) = sprintf "%s/%s" SourceFolder folder
     let Build(folder) = sprintf "%s/%s" BuildFolder folder
+    
+    let ProjFile(project:DotNetProject) =
+        match project with 
+        | Project p -> sprintf "%s/%s/%s.csproj" SourceFolder project.Name project.Name
+        | PrivateProject p ->
+            match p with
+            | Tests -> sprintf "%s/%s/%s.csproj" SourceFolder project.Name project.Name
+            | DocGenerator -> sprintf "%s/CodeGeneration/%s/%s.csproj" SourceFolder project.Name project.Name
+            
 
     let BinFolder(folder) = 
         let f = replace @"\" "/" folder

--- a/build/scripts/Releasing.fsx
+++ b/build/scripts/Releasing.fsx
@@ -8,7 +8,10 @@
 #load @"Versioning.fsx"
 
 open System
+open System.IO
+open System.Linq
 open System.Text
+open System.Xml.Linq
 open Microsoft.FSharp.Quotations
 open Microsoft.FSharp.Quotations.Patterns
 open Fake
@@ -34,8 +37,22 @@ module Release =
 
             let year = sprintf "%i" DateTime.UtcNow.Year
 
-            let jsonDotNetCurrentVersion = "10.0.1"
-            let jsonDotNetNextVersion = "11.0"
+            let jsonDotNetCurrentVersion = 
+                let xName n = XName.op_Implicit n
+                use stream = File.OpenRead <| Paths.ProjFile p
+                let doc = XDocument.Load(stream)
+                let packageReference = 
+                    doc.Descendants(xName "PackageReference")
+                       .FirstOrDefault(fun e -> e.Attribute(xName "Include").Value = "Newtonsoft.Json")
+                if (packageReference <> null) then packageReference.Attribute(xName "Version").Value
+                else String.Empty
+                
+            let jsonDotNetNextVersion = 
+                match jsonDotNetCurrentVersion with
+                | "" -> String.Empty
+                | version -> 
+                    let semanticVersion = SemVerHelper.parse version
+                    sprintf "%i" (semanticVersion.Major + 1) 
 
             let properties =
                 let addKeyValue (e:Expr<string>) (builder:StringBuilder) =
@@ -46,7 +63,9 @@ module Release =
                         | PropertyGet (eo, pi, li) -> (pi.Name, (pi.GetValue(e) |> string))
                         | ValueWithName (obj,ty,nm) -> ((obj |> string), nm)
                         | _ -> failwith (sprintf "%A is not a let-bound value. %A" e (e.GetType()))
-                    builder.AppendFormat("{0}=\"{1}\";", key, value);
+                        
+                    if (isNotNullOrEmpty value) then builder.AppendFormat("{0}=\"{1}\";", key, value)
+                    else builder
                 new StringBuilder()
                 |> addKeyValue <@nextMajorVersion@>
                 |> addKeyValue <@year@>

--- a/build/scripts/Releasing.fsx
+++ b/build/scripts/Releasing.fsx
@@ -34,10 +34,8 @@ module Release =
 
             let year = sprintf "%i" DateTime.UtcNow.Year
 
-            let jsonDotNetVersion = 10
-
-            let jsonDotNetCurrentVersion = sprintf "%i" jsonDotNetVersion
-            let jsonDotNetNextVersion = sprintf "%i" (jsonDotNetVersion + 1)
+            let jsonDotNetCurrentVersion = "10.0.1"
+            let jsonDotNetNextVersion = "11.0"
 
             let properties =
                 let addKeyValue (e:Expr<string>) (builder:StringBuilder) =

